h2/src/main/org/h2/command/Parser.java

```diff
@@ -3137,17 +3137,30 @@ private Expression readTerm() {
                     read("FOR");
                     Sequence sequence = readSequence();
                     r = new SequenceValue(sequence);
-                } else if (equalsToken("TIMESTAMP", name) && readIf("WITH")) {
-                    read("TIME");
-                    read("ZONE");
-                    if (currentTokenType != VALUE
-                            || currentValue.getType() != Value.STRING) {
-                        throw getSyntaxError();
+                } else if (equalsToken("TIMESTAMP", name)) {
+                    if (readIf("WITH")) {
+                        read("TIME");
+                        read("ZONE");
+                        if (currentTokenType != VALUE
+                                || currentValue.getType() != Value.STRING) {
+                            throw getSyntaxError();
+                        }
+                        String timestamp = currentValue.getString();
+                        read();
+                        r = ValueExpression.get(ValueTimestampTimeZone.parse(timestamp));
+                    } else {
+                        if (readIf("WITHOUT")) {
+                            read("TIME");
+                            read("ZONE");
+                        }
+                        if (currentTokenType != VALUE
+                                || currentValue.getType() != Value.STRING) {
+                            throw getSyntaxError();
+                        }
+                        String timestamp = currentValue.getString();
+                        read();
+                        r = ValueExpression.get(ValueTimestamp.parse(timestamp, database.getMode()));
```

Comment:
Plain `TIMESTAMP` and `TIMESTAMP WITHOUT TIME ZONE` literals are now parsed here.